src/com/amazon/ion/impl/_Private_IonBinaryWriterBuilder.java

```diff
@@ -275,6 +282,9 @@ public void setStreamCopyOptimized(final boolean optimized)
 
     //=========================================================================
 
+    public void setBlockSize(int size) {
+        myBinaryWriterBuilder.withUserBlockSize(size);
```

Comment:
Should we also set the symbol table block size?

Reply:
> Should we also set the symbol table block size?

Yes, thanks for catching this. Previously, we only flush the data at `IonRawBinaryWriter` level, and for now we implemented `auto-flush` at the `IonManagedBinaryWriter` level which will flush both symbol table buffer and user writer buffer. In this case, I'm wondering if we should set a separate API to set symbol table block size? The flush operation is decided by the user block boundary check and we might need a different block size for symbol table since when the writer is flushed the symbol table block might be only half occupied, which will lead to the waste of extra bytes.

Reply:
> I think exposing both is fine, but we need to be very clear in the documentation the differences between them and how you would decide whether to change them.

Based on the offline discussion with @tgregg, we should set the symbol table block size matches with the user block size while the size is set smaller than 32K, and keep the symbol table block size as 32k when users set the block size larger than 32K.